hazelcast/src/main/java/com/hazelcast/query/impl/AndResultSet.java

```diff
@@ -161,6 +150,14 @@ public void remove() {
 
     @Override
     public int size() {
-        return setSmallest.size();
+        if (size == -1) {
```

Comment:
It's just iterating over the set once, then it's cached. The only place where it's used instantiated is here:
https://github.com/tombujok/hazelcast/blob/c2472d25552d6ad830858365cc142ef73b4c4a41/hazelcast/src/main/java/com/hazelcast/query/impl/predicates/AndPredicate.java#L99-L99

What do you mean precisely when you say that it's used during evaluation.